packages/administration/src/Component/Datagrid/DatagridManager.php

```diff
@@ -27,4 +30,12 @@ public function createGrid(mixed $name, DataSourceInterface $query): Grid
     {
         return $this->gridFactory->create($name, $query);
     }
+
+    /**
+     * @return \Shopsys\AdministrationBundle\Component\Router\CrudRouteProvider
+     */
+    public function getCrudRouteProvider(): CrudRouteProvider
+    {
+        return $this->crudRouteProvider;
```

Comment:
I am not a big fan of having getters for services that can be autowired where needed :shrug:

Reply:
I will refactor this

Reply:
Ok after refactoring DatagridRowActions Datagrid needs only `Shopsys\FrameworkBundle\Component\Grid\GridFactory` that is now passed by the constructor :) So `DatagridManager` is no longer available